enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+
+Once fully migrated to the new formats, the general flow for updating the catalog's content might look like:
+
+1. The git repository is forked
+2. A bundle image reference is added, updated, or blacklisted in an index config file
+3. A PR is opened against the git repository
+4. An index is built with the patched config, using the current production index as a base
+5. The index content is verified and tested
+6. If tests pass, and the PR is approved, it enters a merge queue
+7. If the production index image changes before it merges, the PR drops from the queue (GOTO 4)
+8. Once merged, build a new production index image
+
+An configuration file will inform a build of the bundles to add or remove, if any, from an index:
+
+```yaml
+from: image://quay.io/operator-framework/community-operators:latest
+included:
+- ref: quay.io/coreos/something-bundle:v1
+  package: something
+  channel: stable
+- ref: quay.io/coreos/something-bundle@sha256:abcdef...
+  package: something
+  channel: alpha
+- ref: quay.io/redhat/something-else-bundle:v5.5
+  package: something-else
+  channel: v5.Y
+exluded:
+- ref: quay.io/redhat/something-else-bundle:v5.4
+  package: something-else
+  channel: v5.Y
+```
+
+A new `-f` option will be added to the `opm index add` subcommand which specifies the path of an index config file. The command will parse the given file and idempotently:
+
+- add bundles referenced by the `included` directive to the index specified by `from`
+- remove bundles referenced by the `excluded` directive from the index specified by `from`
+
+The command will also be capable of sourcing "from indexes" in varying formats using reference prefixes:
+
+- "image://" for container image references
+- "file://" for database file paths
+
+TODO(njhale): staged migration to bundle images
+
+1. create an index configuration specifying the existing catalog image in `from`
+2. reduce the number of manifest versions kept for each operator to the head of each package/channel
+3. walk manifest directories adding to the index database
+4. run `opm index add` using the index config as an overlay
+
+### Migrate away from appregistry
+
+Detailed steps about migration from the pipeline perspective can be found here: https://docs.engineering.redhat.com/display/DELIVERY/New+OCP+Application+Registry%3A+Requirements
+
+### Catalogs for minor OCP versions
+
+As an operator registry maintainer, I need a way to specify what OCP versions are supported by the operators in the registry.
+
+#### Design Proposal
+
+We can use names of branches/tags on dist-git repos as convention for what ocp versions are supported. We can then update each OCP versioned index for pushes to each branch/tag. If this is not possible we can include this in a config file similar to `art.yaml` for openshift operators.
```

Comment:
I think the answer to this question depends on if we want to have a 1:1 relationship between redhat-operator version and openshift version or not. If a matrix of version support is necessary (i.e. version 1.3 of my operator works on Openshift 4.4 4.5 and 4.6) then it may be necessary to have the CI have a more complex registration system (like a yaml file checked into the repository)